app/controllers/assignments_controller.rb

```diff
@@ -35,14 +35,14 @@ def generate_rotation
     user_ids = params.require :user_ids
     start_user = params.require :starting_user_id
     if end_date.before? start_date
-      flash[:errors] = 'The end date must be after the start date.'
-      redirect_back(fallback_location:
-                    roster_assignments_path(@roster)) and return
+      flash.now[:errors] = t('.end_before_start')
+      rotation_generator
```

Comment:
So if `rotation_generator` did a lot of setup for the form that would be hacky to replicate here, I would accept this (theoretically) lesser hack to get things working for now.

That being said, all that method does is set `@start_date` to prepopulate the field for the form. Conveniently we also accept start date as a param above.

Therefore I think it's less weird and even _slightly_ better to just change the `start_date` here to `@start_date` and reuse it for the form.